filament/backend/src/metal/MetalHandles.mm

```diff
@@ -256,9 +268,10 @@ static inline MTLTextureUsage getMetalTextureUsage(TextureUsage usage) {
     PresentDrawableData(const PresentDrawableData&) = delete;
     PresentDrawableData& operator=(const PresentDrawableData&) = delete;
 
-    static PresentDrawableData* create(id<CAMetalDrawable> drawable, MetalDriver* driver) {
+    static PresentDrawableData* create(id<CAMetalDrawable> drawable,
+            std::mutex* drawableMutex, MetalDriver* driver) {
```

Comment:
I really don't like that, passing the lock in here. but I'll let @bejado decide.

Reply:
Not a huge fan of it either, but I didn't see a cleaner way. PresentCallables can potentially stay alive for up to a frame, if the customer's using it to synchronize a Filament frame with a UIKit CATransaction.

One alternative would be making mDrawableMutex a `std::shared_ptr<std::mutex>` and passing it in that form -- that would divorce lifetimes and allow a PresentCallable to cleanly outlive MetalSwapChain if needed. But it seemed wasteful to me to move that mutex into a separate heap allocation and do a bunch of atomics just to get access to another atomic.

(There's also the option of cutting the Gordian Knot, by dispatch_sync()ing from render thread to main thread in order to get the next drawable. But that seems even worse.)